models/common.py

```diff
@@ -323,7 +323,13 @@ def __init__(self, weights='yolov5s.pt', device=torch.device('cpu'), dnn=False,
         super().__init__()
         w = str(weights[0] if isinstance(weights, list) else weights)
         pt, jit, onnx, xml, engine, coreml, saved_model, pb, tflite, edgetpu, tfjs = self.model_type(w)  # get backend
-        w = attempt_download(w)  # download if not local
+        from utils.loggers.wandb.wandb_utils import check_valid_artifact_address, download_model_from_wandb_artifact
+        if check_valid_artifact_address(w):
+            w = download_model_from_wandb_artifact(w)
+            weights = w
+            pt, jit, onnx, xml, engine, coreml, saved_model, pb, tflite, edgetpu, tfjs = self.model_type(w)
+        else:
+            w = attempt_download(w)  # download if not local
```

Comment:
Can you handle this part in the logger files? Feel free to use either `utils/loggers/__init__.py` and `wandb_utils.py`